### DISCUSSION TITLE
Secondary right thumb key activates numeric (not symbol) layer when hold duration is short

### DISCUSSION THREAD

**Discussion by c-mart** [manna-harbour/miryoku#314] [Q&A] [answered]
On ZMK here. Holding the secondary thumb key on right hand (the innermost key by default) _should_ activate the symbol layer, but for shorter hold durations (less than approximately half a second), it activates the numeric layer instead. For example, if I try to type `%`, and don't pause for long enough after pressing the layer switch key, I get `5` instead.

It doesn't seem to happen with the opposite hand (left secondary thumb activating mouse layer on right hand). I also don't have this problem with the primary (middle by default) or tertiary (outermost) thumb key on either hand.

I confirmed this continues to happen:

- with an un-customized GitHub Workflow build (configuring only the workflow options for my board and shield)
- with my usual customization (in https://github.com/c-mart/miryoku_zmk/tree/cmart)
- with the MIRYOKU_KLUDGE_THUMBCOMBOS code removed. (I wasn't certain that it's related, just tried as a hunch.)

Anyone else encounter this?

**Comment by manna-harbour** [comment, marked as answer]
Can you test with a different host environment? Also try https://github.com/manna-harbour/miryoku_zmk#tap-delay and see the related ZMK issues e.g. https://github.com/zmkfirmware/zmk/issues/1444, https://github.com/zmkfirmware/zmk/issues/1443, https://github.com/zmkfirmware/zmk/issues/759.

As a work-around you can use Shift on Num.

**Reply by c-mart** [reply]
Ah, I do _not_ have this problem when the keyboard outputs to my Android phone. My computer runs GNOME with Wayland (Debian 12), and [GNOME issue #5890](https://gitlab.gnome.org/GNOME/gnome-shell/-/issues/5890) (linked in https://github.com/zmkfirmware/zmk/issues/759) sounds exactly like the problem. So, this is fundamentally a host issue, not a Miryoku or ZMK issue.

Unfortunately [my workflow builds fail](https://github.com/c-mart/miryoku_zmk/actions/runs/8606665159/job/23585558828) when I add `#define MIRYOKU_KLUDGE_TAPDELAY` to my `custom_config.h`, with `devicetree error: /keymap/BASE: undefined node label 'u_tapdelay_lt_SPACE'`.

Fortunately, the workaround in that GNOME issue (`pkill ibus` if you don't need [it](https://en.wikipedia.org/wiki/Intelligent_Input_Bus)) is an effective fix for me. It's also been fixed upstream (so it looks like Ubuntu 24.04 will get it, and maybe Debian 13 when it arrives).

I'm good here :slightly_smiling_face: Thank you for the pointers @manna-harbour.

**Reply by manna-harbour** [reply]
Great!

> Unfortunately [my workflow builds fail](https://github.com/c-mart/miryoku_zmk/actions/runs/8606665159/job/23585558828) when I add #define MIRYOKU_KLUDGE_TAPDELAY to my custom_config.h, with devicetree error: /keymap/BASE: undefined node label 'u_tapdelay_lt_SPACE'.

Thanks, that's fixed now. Although the default macro timings have changed in ZMK, so it may no longer effective, and may need updating. If you'd like to use it and it doesn't work, let me know.
